Assets/MixedRealityToolkit.Services/InputSystem/MixedRealityInputSystem.cs

```diff
@@ -843,17 +868,18 @@ public void RaiseSourcePoseChanged(IMixedRealityInputSource source, IMixedRealit
         /// <inheritdoc />
         public void RaisePreFocusChanged(IMixedRealityPointer pointer, GameObject oldFocusedObject, GameObject newFocusedObject)
```

Comment:
With the current implementation it seems that we can dispatch the same event up to three times to global event listeners. We should do that just once, at the beginning of the function. Same goes for `RaiseFocusChanged`.

Reply:
It would depend on a usecase we want to support. These 3 events are invoked on different objects. Would we ever want information about focus receiver passed to a global listener? E.g. some debug inspector for focus events might want to receive all 3 copies of an event.

Reply:
I changed this to being called just once on the global listeners. Since the FocusEventData effectively contains all 3 possible receivers, you can read all three if you want.